cpp/include/IceUtil/PushDisableWarnings.h

```diff
@@ -15,6 +15,7 @@
 #    pragma warning(disable : 4251) // class ... needs to have dll-interface to be used by clients of class ...
 #    pragma warning(disable : 4512) //  ... assignment operator could not be generated
 #    pragma warning(disable : 4275) // non dll-interface class ... used as base for dll-interface class ...
+#    pragma warning(disable : 4996) // Allows Slice definitions to reference other deprecated Slice definitions.
```

Comment:
This is necessary to support things like:
```
["deprecated"] MyStruct { bool b; }
class MyClass { MyStruct ms; }
```
Without these extra `pragma-disable`s, the generated code would emit deprecation warnings because
`MyClass references a deprecated type: MyStruct`.

In the future, we'll have the Slice compiler itself emit warnings for when you use deprecated Slice definitions, so no point emitting double-warnings IMO. Especially because this is generated code that the user has no control over.